docs/clients/promtail/troubleshooting.md

```diff
@@ -98,11 +98,9 @@ from there. This means that if new log entries have been read and pushed to the
 ingester between the last sync period and the crash, these log entries will be
 sent again to the ingester on `promtail` restart.
 
-However, for each log stream (set of unique labels) the Loki ingester skips all
-log entries received out of timestamp order. For this reason, even if duplicated
-logs may be sent from `promtail` to the ingester, entries whose timestamp is
-older than the latest received will be discarded to avoid having duplicated
-logs. To leverage this, it's important that your `pipeline_stages` include
-the `timestamp` stage, parsing the log entry timestamp from the log line instead
-of relying on the default behaviour of setting the timestamp as the point in
-time when the line is read by `promtail`.
+However, for each log stream (set of unique labels), the Loki ingester will
+reject all log lines received out of order. If files read by `promtail` have a
+timestamp in the log line, it's important for the `pipeline_stages` to include a
+`timestamp` stage to parse this line to avoid out of error problems. Read the
+[Loki Overview documentation](../../overview/README.md#timestamp-ordering) page
+for more information on how timestamp ordering is performed.
```

Comment:
really it's less likely for out of error problems if you avoid the timestamp stage...  promtail would use its own clock and almost guarantee never sending logs out of order.  By introducing the timestamp stage you are at the mercy of whatever created the logs and we know some log sources log things out of order.

I actually think the original intent of this block was to point out that if promtail crashes or restarts or something causes old logs to be resent, the old logs that were already in loki will be ignored and only new logs accepted, but this only happens if you have a pipeline_stage reading the timestamp from the file.